aiida_cp2k/calculations/__init__.py

```diff
@@ -167,6 +168,21 @@ def prepare_for_submission(self, folder):
                                self.inputs.structure if 'structure' in self.inputs else None)
             write_basissets(inp, self.inputs.basissets, folder)
 
+        # if we have both basissets and structure we can start helping the user :)
+        if 'basissets' in self.inputs and 'structure' in self.inputs:
+            try:
+                scf_section = inp.get_section_dict('FORCE_EVAL/DFT/SCF')
+
+                if 'SMEAR' in scf_section and 'ADDED_MOS' not in scf_section:
+                    # now is our time to shine!
+                    added_mos = estimate_added_mos(self.inputs.basissets, self.inputs.structure)
+                    inp.add_keyword('FORCE_EVAL/DFT/SCF/ADDED_MOS', added_mos)
+                    self.logger.info(f'The FORCE_EVAL/DFT/SCF/ADDED_MOS was added with an automatically estimated value'
+                                     f' of {added_mos}')
+
+            except (KeyError, TypeError):  # no SCF, no smearing, or multiple FORCE_EVAL, nothing to do (yet)
+                pass
```

Comment:
I think a nice message saying something like "I tried to add some molecular orbitals, but did not succeed" would be very useful.

Reply:
Ok, the try/except around the whole section was probably a bit misleading. I've corrected that.